thavalon-server/src/database/games/db_game.rs

```diff
@@ -0,0 +1,282 @@
+                player_id,
+                self._id,
+                self.status
+            );
+            return Err(DBGameError::InvalidStateError);
+        }
+
+        self.players.insert(player_id);
+        self.display_names.insert(display_name);
+        let update_doc = doc! {
+            "$set": {
+                "players": bson::to_bson(&self.players).unwrap(),
+                "display_names": bson::to_bson(&self.display_names).unwrap()
+            }
+        };
+
+        self.update_db(update_doc).await
+    }
+
+    /// Removes a player to the DB game instance, updating the DB accordingly.
+    /// Players can only be removed if the game status is `Lobby`.
+    ///
+    /// # Arguments
+    ///
+    /// * `player_id` - The player ID to add to the game
+    /// * `display_name` - The display name of the joining player
+    ///
+    /// # Returns
+    ///
+    /// * Empty type on success
+    /// * `DBGameError::InvalidStateError` if the game state isn't `Lobby`
+    /// * `DBGameError::UpdateError` if a DB update fails
+    pub async fn remove_player(
+        &mut self,
+        player_id: &String,
+        display_name: &String,
+    ) -> Result<(), DBGameError> {
+        log::info!("Removing player {} to game {}.", player_id, self._id);
+        if self.status != DBGameStatus::Lobby {
+            log::error!(
+                "ERROR: attempted to remove player {} to game {} while in state {:?}. 
+            Players may only be removed during the Lobby phase.",
+                player_id,
+                self._id,
+                self.status
+            );
+            return Err(DBGameError::InvalidStateError);
+        }
+
+        self.players.remove(player_id);
+        self.display_names.remove(display_name);
+        let update_doc = doc! {
+            "$set": {
+                "players": bson::to_bson(&self.players).unwrap(),
+                "display_names": bson::to_bson(&self.display_names).unwrap()
+            }
+        };
+
+        self.update_db(update_doc).await
+    }
+
+    /// Helper function to get a handle to the game collection.
+    ///
+    /// # Returns
+    ///
+    /// `Collection` of Thavalon DB games
+    async fn get_collection() -> Collection {
+        get_database().await.collection(GAME_COLLECTION)
+    }
+
+    /// Updates the game database using the provided update document.
+    ///
+    /// # Arguments
+    ///
+    /// * `update_doc` - The document with fields to update
+    ///
+    /// # Returns
+    ///
+    /// * Empty type on success, `DBGameError` on failure
+    async fn update_db(&self, update_doc: Document) -> Result<(), DBGameError> {
+        let collection = DatabaseGame::get_collection().await;
+        if let Err(e) = collection
+            .update_one(doc! {"_id": &self._id}, update_doc, None)
+            .await
+        {
+            log::error!(
+                "ERROR: failed to update database game while starting the game. {}.",
```

Comment:
Could you drop "while starting the game" since that's not the only time we call this?